middleware/grpc_instrumentation.go

```diff
@@ -173,18 +172,83 @@ func (s *instrumentedClientStream) CloseSend() error {
 	return err
 }
 
-// errorCode converts an error into an error code string.
-func errorCode(err error) string {
-	if err == nil {
-		return "2xx"
+type InstrumentationOption func(*instrumentationLabel)
+
+var (
+	// ReportGRPCStatusOption is an InstrumentationOption that is used for enabling gRPC status codes to be used
+	// in instrumentation labels.
+	ReportGRPCStatusOption InstrumentationOption = func(instrumentationLabel *instrumentationLabel) {
+		instrumentationLabel.reportGRPCStatus = true
+	}
+)
+
+func applyInstrumentationOptions(maskHTTPStatuses bool, options ...InstrumentationOption) instrumentationLabel {
+	instrumentationLabel := instrumentationLabel{maskHTTPStatus: maskHTTPStatuses}
+	for _, opt := range options {
+		opt(&instrumentationLabel)
+	}
+	return instrumentationLabel
+}
+
+type instrumentationLabel struct {
+	reportGRPCStatus bool
+	maskHTTPStatus   bool
+}
+
+// getInstrumentationLabel converts an error into an error code string by applying the configurations
+// contained in this instrumentationLabel object.
+func (i *instrumentationLabel) getInstrumentationLabel(err error) string {
+	statusCode := errorToStatusCode(err)
+	return i.statusCodeToString(statusCode)
+}
+
+func (i *instrumentationLabel) statusCodeToString(statusCode codes.Code) string {
+	if isHTTPStatusCode(statusCode) {
+		statusFamily := int(statusCode / 100)
+		if i.maskHTTPStatus {
+			return strconv.Itoa(statusFamily) + "xx"
+		}
+		return strconv.Itoa(int(statusCode))
 	}
 
-	if errResp, ok := httpgrpc.HTTPResponseFromError(err); ok {
-		statusFamily := int(errResp.Code / 100)
-		return strconv.Itoa(statusFamily) + "xx"
-	} else if grpcutil.IsCanceled(err) {
+	if i.reportGRPCStatus {
+		return statusCode.String()
+	}
+
+	if statusCode == codes.OK {
+		if i.maskHTTPStatus {
+			return "2xx"
+		}
+		return "success"
+	}
+
+	if statusCode == codes.Canceled {
 		return "cancel"
-	} else {
-		return "error"
 	}
+
+	return "error"
+}
+
+// errorToStatusCode extracts a status code from the given error, and does the following:
+//
+//   - If the error is nil, codes.OK is returned.
+//   - If the error corresponds to context.Canceled, codes.Canceled is returned.
+//   - Otherwise, the actual status code of the error is returned.
+func errorToStatusCode(err error) codes.Code {
```

Comment:
nit: This description of implementation seems unnecessary. Same for next function. (I'd only comment public functions, of functions with tricky behaviour. But this is pretty straightforward)